File: go/FindDiff/FindDiff.go
package main

import "fmt"

// Time -> 0(N) Space -> 0(1)
func findTheDifference(s, t string) byte {
	lookup := make(map[byte]int, 0)

	for idx := range s {
		lookup[s[idx]]++
	}

	for idx := range t {
		if val, exist := lookup[t[idx]]; exist && val > 0 {
			lookup[t[idx]]--
		} else {
			return t[idx]
		}
	}

	return 0
}

func main() {
	fmt.Println(findTheDifference("abcd", "abcde"))
	fmt.Println(findTheDifference("", "y"))
}

File: python/findDiff.py
# Time -> 0(N) Space -> 0(1)
def findTheDifference(s: str, t: str) -> str:
    lookup = {}

    for val in s:
        lookup[val] = lookup.get(val, 0) + 1

    for val in t:
        if val in lookup and lookup[val] > 0:
            lookup[val] -= 1
        else:
            return val

    return ""
